govcd/disk.go

```diff
@@ -0,0 +1,283 @@
+	if err = decodeBody(resp, disk.Disk); err != nil {
+		return nil, fmt.Errorf("error decoding create disk params response: %s", err)
+	}
+
+	// Return the disk
+	return disk, nil
+}
+
+// Update an independent disk
+// 1 Verify that the disk is not attached to a virtual machine.
+// 2 Use newDiskInfo to change update the independent disk.
+// 3 Return task of independent disk update
+// Please verify the independent disk is not connected to any VM before calling this function.
+// If the independent disk is connected to a VM, the task will be failed.
+// Reference: vCloud API Programming Guide for Service Providers vCloud API 30.0 PDF Page 104 - 106,
+// https://vdc-download.vmware.com/vmwb-repository/dcr-public/1b6cf07d-adb3-4dba-8c47-9c1c92b04857/
+// 241956dd-e128-4fcc-8131-bf66e1edd895/vcloud_sp_api_guide_30_0.pdf
+func (d *Disk) Update(newDiskInfo *types.Disk) (Task, error) {
+	var err error
+	var execLink *types.Link
+
+	// Find the proper link for request
+	for _, diskLink := range d.Disk.Link {
+		if diskLink.Rel == types.RelEdit && diskLink.Type == types.MimeDisk {
+			execLink = diskLink
+			break
+		}
+	}
+
+	if execLink == nil {
+		return Task{}, fmt.Errorf("exec link not found")
+	}
+
+	// Parse request URI
+	reqUrl, err := url.ParseRequestURI(execLink.HREF)
+	if err != nil {
+		return Task{}, fmt.Errorf("error parse URI: %s", err)
+	}
+
+	// Prepare the request payload
+	xmlPayload, err := xml.Marshal(&types.Disk{
+		Xmlns:          types.NsVCloud,
+		Description:    newDiskInfo.Description,
+		Size:           newDiskInfo.Size,
+		Name:           newDiskInfo.Name,
+		StorageProfile: newDiskInfo.StorageProfile,
+		Owner:          newDiskInfo.Owner,
+	})
+	if err != nil {
+		return Task{}, fmt.Errorf("error xml.Marshal: %s", err)
+	}
+
+	// Send request
+	req := d.client.NewRequest(nil, http.MethodPut, *reqUrl, bytes.NewBufferString(xml.Header+string(xmlPayload)))
+	req.Header.Add("Content-Type", execLink.Type)
+	resp, err := checkResp(d.client.Http.Do(req))
+	if err != nil {
+		return Task{}, fmt.Errorf("error find disk: %s", err)
+	}
+
+	// Decode response
+	task := NewTask(d.client)
+	if err = decodeBody(resp, task.Task); err != nil {
+		return Task{}, fmt.Errorf("error decoding find disk response: %s", err)
+	}
+
+	// Return the task
+	return *task, nil
+}
+
+// Remove an independent disk
+// 1 Delete the independent disk. Make a DELETE request to the URL in the rel="remove" link in the Disk.
+// 2 Return task of independent disk deletion.
+// Please verify the independent disk is not connected to any VM before calling this function.
+// If the independent disk is connected to a VM, the task will be failed.
+// Reference: vCloud API Programming Guide for Service Providers vCloud API 30.0 PDF Page 106 - 107,
+// https://vdc-download.vmware.com/vmwb-repository/dcr-public/1b6cf07d-adb3-4dba-8c47-9c1c92b04857/
+// 241956dd-e128-4fcc-8131-bf66e1edd895/vcloud_sp_api_guide_30_0.pdf
+func (d *Disk) Delete() (Task, error) {
```

Comment:
I think check regarding "Please verify the independent disk is not connected to any VM before calling this function." should be this function scope.